__tests__/helpers.js

```diff
@@ -0,0 +1,47 @@
+"use strict";
+
+const MemoryFileSystem = require("memory-fs");
+const webpack = require("webpack");
+
+exports.PluginEnvironment = class PluginEnvironment {
+	constructor() {
+		this.events = [];
+	}
+
+	getEnvironmentStub() {
+		return {
+			plugin: (name, handler) => {
+				this.events.push({
+					name,
+					handler
+				});
+			}
+		};
+	}
+
+	getEventBindings() {
+		return this.events;
+	}
+};
+
+
+exports.runCompiler = function runCompiler(compiler) {
```

Comment:
Well, I'm ok with either direction. I hadn't added many snapshots tests yet - but I built this with the idea of making many in mind and thought it would be useful.